simulation/src/sum_reduce.cpp

```diff
@@ -3,12 +3,13 @@
 using namespace pcl::simulation;
 
 pcl::simulation::SumReduce::SumReduce(int width, int height, int levels)
-: levels_(levels), width_(width), height_(height)
+: levels_(levels)
+, width_(width)
+, height_(height)
+, sum_program_(new gllib::Program())
```

Comment:
std::make_shared should be better option here.
std::make_shared often used to create
shared_ptr objects for both efficiency and exception-safety reasons. make_shared can allow memory for control block and managed object to
be allocated together in single memory allocation

Reply:
In general, you are correct about the single memory allocation. But here, `sum_program_` is a class member, so if you use make_shared, you would have to copy the returned shared_ptr to `sum_program_`. And then, I don't think the control block and the managed object will stay together?
Not sure what you mean with exception-safety. The allocation in `new` could fails of course, but I guess then an exception is inevitable?